microsoft_cve_rag/application/etl/transformer.py

```
@@ -1184,121 +2081,404 @@ def transform_patch_posts(patch_posts: List[Dict[str, Any]], process_all: bool =
+            # Group documents by normalized subject for better organization
+            subject_groups = {}
+            for _, row in final_df.iterrows():
+                thread_id = row["metadata"].get("thread_id")
+                if thread_id not in subject_groups:
+                    subject_groups[thread_id] = []
+                # Convert row to dict and ensure metadata is a dict
+                row_dict = row.to_dict()
+                if isinstance(row_dict["metadata"], str):
+                    try:
+                        row_dict["metadata"] = json.loads(row_dict["metadata"])
+                    except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)